src/gui_common/MouseHoverPanel.cs

```diff
@@ -0,0 +1,197 @@
+﻿using Godot;
+using Godot.Collections;
+
+/// <summary>
+///   The panel that shows what the player is hovering over/inspecting.
+/// </summary>
+public class MouseHoverPanel : PanelContainer
+{
+    [Export]
+    public NodePath CategoriesContainerPath = null!;
+
+    [Export]
+    public NodePath NothingHereContainerPath = null!;
+
+#pragma warning disable CA2213 // Disposable fields should be disposed
+    private Container categoriesContainer = null!;
+    private Container nothingHereContainer = null!;
+#pragma warning restore CA2213 // Disposable fields should be disposed
+
+    private System.Collections.Generic.Dictionary<string, MouseHoverCategory> categories = new();
+    private Array categoriesOrdered = new();
+
+    public override void _Ready()
+    {
+        categoriesContainer = GetNode<Container>(CategoriesContainerPath);
+        nothingHereContainer = GetNode<Container>(NothingHereContainerPath);
+    }
+
+    public override void _Process(float delta)
+    {
+        var visibleEntriesCount = 0;
+
+        MouseHoverCategory? firstVisibleCategory = null;
+
+        foreach (MouseHoverCategory category in categoriesOrdered)
+        {
+            category.Visible = category.TotalEntriesCount > 0;
+            category.SeparatorVisible = true;
```

Comment:
This could be refactored to be `firstVisibleCategory != null` to avoid having to modify this variable twice (the second modification being after the loop).